### ISSUE TITLE
How to run fine-tuning on VQA2 dataset?

### ISSUE THREAD

**Issue by TopCoder2K** [ashkamath/mdetr#43]
Experiments with VQA v.2 dataset are described in Appendix E of the article. But it's not clear from [main.py](https://github.com/ashkamath/mdetr/blob/0b747b99e2995c3c429f1391cb8e6104eaec7f21/main.py) and [run_with_submitit.py](https://github.com/ashkamath/mdetr/blob/main/run_with_submitit.py) files how to run the fine-tuning (I've tried to write the same command that is used for [fine-tuning on CLEVR](https://github.com/ashkamath/mdetr/blob/main/.github/gqa.md)). I've also found [vqa_coco_format.py](https://github.com/ashkamath/mdetr/blob/main/scripts/fine-tuning/vqa_coco_format.py) but it seems like preparation of the data, not fine-tuning itself. Also, I've encountered using [build_dataset](https://github.com/ashkamath/mdetr/blob/0b747b99e2995c3c429f1391cb8e6104eaec7f21/datasets/__init__.py#L33) function in [main.py](https://github.com/ashkamath/mdetr/blob/0b747b99e2995c3c429f1391cb8e6104eaec7f21/main.py) and I don't see VQA v2 in the function :(
Could you please explain how to do so?

UPD 1 (09.27.21):
I've downloaded COCO and VQA v2 datasets and ran 
```
python scripts/fine-tuning/vqa_coco_format.py --data_path VQA_v2_dataset/ --img_path COCO_dataset/images/ --coco_path COCO_dataset/
```
And the processing has finished correctly. Now I'm thinking how to write VQA v2 dataset script... 

UPD 2 (10.03.21)
It seems I managed to implement all the necessary classes and fix the code. I'm currently doing an experiment eval -> train on vqa2 -> eval. As soon as it successfully finishes I'll push the code into my fork of the repo.

UPD 3 (10.03.21)
Yeah, it works! Here is the link: https://github.com/TopCoder2K/mdetr. I haven't written any documentation because I'm not sure that fine-tuning on VQA is useful to anybody.)) If you have any question, please ask here :)

**Comment by TopCoder2K** [comment]
Good afternoon, @ashkamath!

Thank you for the great MDETR! 
I've been trying to learn MDETR on the VQA2 dataset. As I wrote above, I managed to implement the code that is needed to run training on the VQA2. Then I decided to conduct your experiment from the Appendix E of the article. I fine-tuned on the GQA balanced with `--no-detection` option for 10 epochs and then fine-tuned on the VQA2 for 25 epochs. But the results are quite strange. It seems that the model hasn't learned, the loss on the GQA has increased and on the VQA almost hasn't changed. Evaluation during training was performed on the `val` and `minival` splits of the GQA and VQA2 respectively.
![image](https://user-images.githubusercontent.com/47208659/136693855-5767b400-5cda-46ac-8238-699621eee547.png)
![image](https://user-images.githubusercontent.com/47208659/136693865-0e1707b1-d1cb-42a9-9d0a-a0d1463e2b26.png)

Here are the commands I've used.
Fine-tuning on the GQA:
```
python run_with_submitit.py --dataset_config configs/gqa.json --ngpus 1 --nodes 1 --ema --epochs 10 --epoch_chunks 25 --do_qa --split_qa_heads --lr_drop 150 --resume https://zenodo.org/record/4721981/files/pretrained_resnet101_checkpoint.pth --batch_size 4 --no_detection --qa_loss_coef 25 --lr 1.4e-4 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5 --output-dir ~/MDETR/mdetr/checkpoint
```
Fine-tuning on the VQA2:
```
python run_with_submitit.py --dataset_config configs/vqa2.json --ngpus 1 --nodes 1  --epochs 25 --epoch_chunks 25 --do_qa --split_qa_heads --lr_drop 150 --backbone resnet101 --load ~/MDETR/mdetr/checkpoint/pchelintsev/experiments/19311/BEST_checkpoint.pth --batch_size 4 --no_aux_loss --no_contrastive_align_loss --no_detection --qa_loss_coef 25 --lr 1.4e-4 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5 --output-dir ~/MDETR/mdetr/checkpoint --do_qa_with_qa_fine-tuned
```
Evaluation on the VQA2:
```
python main.py --dataset_config configs/vqa2.json --eval --do_qa --split_qa_heads --no_contrastive_align_loss --no_aux_loss --no_detection --backbone resnet101 --qa_loss_coef 25 --resume ~/MDETR/mdetr/checkpoint/pchelintsev/experiments/5063/BEST_checkpoint.pth
```

The only significant difference is that I did 10 epochs on the GQA balanced while you did 5 epochs on the GQA all. But can it have such an impact? I think that hyperparameters can be wrong... Could you please provide what options and what hyperparameters values you used?

**Comment by TopCoder2K** [comment]
I've tried the second option by running the following command: 
`python run_with_submitit.py --dataset_config configs/vqa2.json --ngpus 1 --nodes 1 --ema --epochs 10 --epoch_chunks 25 --do_qa --split_qa_heads --load pretrained_resnet101_checkpoint.pth --batch_size 4 --no_aux_loss --no_detection --lr 7e-5 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5 --output-dir ~/MDETR/mdetr/checkpoint`
Here I've tried to adjust it according to your command 
```
python run_with_submitit.py --backbone "resnet101" --dataset_config configs/vqa2.json --num_queries 100 --batch_size 4 --num_workers 5 --schedule linear_with_warmup --text_encoder_type roberta-base --ngpus 8 --nodes 4 --ema --do_qa --load path/to/gqa/model --no_aux_loss --no_detection --lr 7e-5 --lr_backbone 1.4e-5 --text_encoder_lr 7e-5 --bce_qa --epochs 25 
```
So, I have deleted `qa_coeff`, changed `lr`, removed `lr_drop`, turned on `contrastive_align_loss`.
Unfortunately, the results are the not what I would like.
![image](https://user-images.githubusercontent.com/47208659/139530631-88452720-4af4-437b-9b68-7a15f83557a3.png)

Could you please post the exact command you used?
